packages/ckeditor5-dev-env/lib/release-tools/tasks/update-package-versions.js

```diff
@@ -0,0 +1,281 @@
+ * @param {Boolean} dryRun If set to true, diff of changes that would be made is calculated, and included in the returned object. Without
+ * this flag, file is updated normally.
+ * @returns {UpdateResult}
+ */
+function updateDirectory( pathToUpdate, dryRun ) {
+	const globPattern = pathToUpdate + '/*/package.json';
+	const packageJsonArray = glob.sync( globPattern );
+
+	if ( !packageJsonArray.length ) {
+		return { found: 0, updated: 0, differences: [] };
+	}
+
+	let updatedFiles = 0;
+	const differences = [];
+
+	for ( const file of packageJsonArray ) {
+		const currentFileData = fs.readFileSync( file, 'utf-8' );
+		const parsedData = JSON.parse( currentFileData );
+
+		updateObjectProperty( parsedData, 'dependencies' );
+		updateObjectProperty( parsedData, 'devDependencies' );
+
+		const newFileData = JSON.stringify( parsedData, null, 2 ) + '\n';
+
+		if ( currentFileData !== newFileData ) {
+			updatedFiles++;
+
+			if ( dryRun ) {
+				differences.push( {
+					file,
+					content: diff( currentFileData, newFileData, { newlineIsToken: true } )
+				} );
+			} else {
+				fs.writeFileSync( file, newFileData, 'utf-8' );
+			}
+		}
+	}
+
+	return { found: packageJsonArray.length, updated: updatedFiles, differences };
+}
+
+/**
+ * Updates the CKEditor 5 dependencies, (with the exception of `-dev`, `-inspector`, `-react`, `-vue` and `-angular`), in the provided
+ * `package.json` file.
+ *
+ * @param {Object} parsedPkgJson Object to update.
+ * @param {String} propertyName Name of the property to update.
+ */
+function updateObjectProperty( parsedPkgJson, propertyName ) {
+	const version = parsedPkgJson.version;
+	const regex = /^@ckeditor\/ckeditor5-([a-z]+)|^ckeditor5$/;
+	const exceptions = [
+		'dev',
+		'inspector',
+		'react',
+		'vue',
+		'angular'
+	];
+
+	for ( const dependency in parsedPkgJson[ propertyName ] ) {
+		const match = dependency.match( regex );
+
+		if ( !match || exceptions.includes( match[ 1 ] ) ) {
+			continue;
+		}
+
+		parsedPkgJson[ propertyName ][ dependency ] = `^${ version }`;
+	}
+}
+
+/**
+ * Takes in raw changelog for a single file generated by `diff` library, and returns formatted array of strings, containing
+ * human-readable, line by line changelog of a file, with removals and additions colored. If the file has any longer parts without changes,
+ * these will be partially hidden.
+ *
+ * @param {Array<Object>} diff Array of changes for a single file generated by `diff` library.
+ * @returns {Array<String>} Formatted changelog split into single lines.
+ */
+function formatDiff( diff ) {
+	const formattedDiff = [];
+	const regex = /(?<=":) (?=")/;
+
+	for ( let i = 0; i < diff.length; i++ ) {
+		const previous = diff[ i - 1 ];
+		const current = diff[ i ];
+		const next = diff[ i + 1 ];
+		const currentLines = current.value.split( '\n' );
+
+		const validReplaceSequence = current.removed && next.added && regex.test( current.value ) && regex.test( next.value );
```

Comment:
More or less. I've changed `Display` to `Format`, because the diff will be displayed either way, but if this check is passed, it will merge them into a single line for readability, or _pretty-print_ them if you will.